src/java/io/bazel/rulesscala/scalac/ScalacProcessor.java

```diff
@@ -179,27 +177,50 @@ public void processRequest(List<String> args) throws Exception {
   }
 
   private static boolean matchesFileExtensions(String fileName, String[] extensions) {
-    for (String e: extensions) {
+    for (String e : extensions) {
       if (fileName.endsWith(e)) {
         return true;
       }
     }
     return false;
   }
 
+  private static String[] encodeBazelTargets(String[] targets) {
+    return Arrays.stream(targets)
+            .map(label -> label.replace(":", ";"))
+            .toArray(String[]::new);
+  }
+
   private static void compileScalaSources(CompileOptions ops, String[] scalaSources, Path tmpPath) throws IllegalAccessException {
+    String[] targets = encodeBazelTargets(ops.indirectTargets);
+
+    String[] pluginParamsInUse = {
+      "-P:dependency-analyzer:direct-jars:" + String.join(":", ops.directJars),
+      "-P:dependency-analyzer:indirect-jars:" + String.join(":", ops.indirectJars),
+      "-P:dependency-analyzer:indirect-targets:" + String.join(":", targets),
+    };
+
+    String[] pluginParams;
+    if (ops.enableDependencyAnalyzer) {
+      pluginParams = pluginParamsInUse;
```

Comment:
can we move 197 here to make it easier to verify that when we don't use the dependency analyzer we won't use the options.